atcoder-problems-frontend/src/pages/UserPage/Recommendations.tsx

```diff
@@ -18,11 +18,40 @@ import {
   predictSolveProbability,
   predictSolveTime
 } from "../../utils/ProblemModelUtil";
-import { Button, ButtonGroup, Row } from "reactstrap";
+import { 
+  Button, 
+  ButtonGroup, 
+  DropdownItem,
+  DropdownMenu,
+  DropdownToggle,
+  Row,
+  UncontrolledDropdown,
+} from "reactstrap";
 import HelpBadgeTooltip from "../../components/HelpBadgeTooltip";
 import ProblemLink from "../../components/ProblemLink";
 
-const RECOMMEND_NUM = 10;
+const RECOMMEND_NUM_OPTIONS = [
+  {
+    text: '10',
+    value: 10,
+  },
+  {
+    text: '20',
+    value: 20,
+  },
+  {
+    text: '50',
+    value: 50,
+  },
+  {
+    text: '100',
+    value: 100,
+  },
+  {
+    text: 'All',
+    value: -1,
```

Comment:
When the option is `All`, `recommendNum` will be `-1`. `.slice(0, -1)` will return the list of the all the element except the last one. For example, `List([0, 1, 2, 3]).slice(0, -1) == [0, 1, 2]`. Since `.slice(0, INF)` will return unmodified full list, it's better to set `INF` for `All`.

Reply:
Yep, silly bug